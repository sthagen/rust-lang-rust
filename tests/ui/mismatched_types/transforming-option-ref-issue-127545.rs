// Regression test for <https://github.com/rust-lang/rust/issues/127545>.
#![crate_type = "lib"]

pub fn foo(arg: Option<&Vec<i32>>) -> Option<&[i32]> {
    arg //~ ERROR 5:5: 5:8: mismatched types [E0308]
}
