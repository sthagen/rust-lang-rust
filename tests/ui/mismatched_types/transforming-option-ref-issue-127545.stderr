error[E0308]: mismatched types
  --> $DIR/transforming-option-ref-issue-127545.rs:5:5
   |
LL | pub fn foo(arg: Option<&Vec<i32>>) -> Option<&[i32]> {
   |                                       -------------- expected `Option<&[i32]>` because of return type
LL |     arg
   |     ^^^ expected `Option<&[i32]>`, found `Option<&Vec<i32>>`
   |
   = note: expected enum `Option<&[i32]>`
              found enum `Option<&Vec<i32>>`
help: try using `.map(|v| &**v)` to convert `Option<&Vec<i32>>` to `Option<&[i32]>`
   |
LL |     arg.map(|v| &**v)
   |        ++++++++++++++

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0308`.
